[24기 박은정] [Mission 1~4] 로그인, 메인페이지 구현완료

eunjeong-97 opened the pull request:
- 로그인 레이아웃 완료
- 로그인 유효성검사 로직완료
- 메인 페이지 레이아웃 완료
- 메인 페이지 댓글기능 구현완료

>wecode PR message template 및 체크 리스트 입니다. 
아래 사항들을 전부 작성/체크 하시고 PR 해주세요!

## 수정 사항 간략한 한줄 요약
(여기에 수정 사항에 대한 간략한 한줄 요약/제목 작성해 주세요.)
- 로그인, 메인페이지 구현완료
		 
## 수정 사항들 자세한 내용
(여기에 수정 사항에 대한 자세한 내용을 작성해 주세요.)
- id '@' 포함, pw 5 글자 이상 입력 시 버튼 활성화(예시)

## 기타 질문 및 특이 사항
(궁금한 사항들, 하면서 느낀 점들 공유해주실 것이 있으면 자유롭게 작성해 주세요.)
- html 파일에 div태그가 많다고 생각되는데 줄일 방법이 있을까요?
- img 태그에 class명을 줘서 너비나 높이값을 조절하려고 하면 작동되지 않아 img태그를 감싸는 div태그를 만들어서 div요소에 스타일 적용했습니다
- git add/commit/push 했는데 깃허브페이지에서 보면 PR하는 버튼이 생성되지 않습니다
- PR버튼이 따로 생성되지 않아서 feature/eunjeongPark 저장소에 들어와 직접 PR요청을 했는데 크게 문제가 없는지 궁금합니다

## 체크 리스트 (아래 사항들이 전부 체크되어야만 merge가 됩니다!)
- [ x ] 필요한 test들을 완료하였고 기능이 제대로 실행되는지 확인 하였습니다.
- [ x ] Wecode의 코드 스타일 가이드에 맞추어 코드를 작성 하였습니다.
- [ x ] 제가 의도한 파일들과 수정 사항들만 커밋이 된 것을 확인 하였습니다.
- [ x ] 본 수정 사항들을 팀원들과 사전에 상의하였고 팀원들 모두 해당 PR에 대하여 알고 있습니다.

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+            <div class="avatar">
+              <img alt="avatar" src="./img/avatar-yeonuk.jpg" />
+            </div>
+            <div class="friendsname">
```
camelCase로 변경해주세요 :)

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+        <article class="feed">
+          <header class="feedHeader">
+            <div class="avatar">
+              <img alt="avatar" src="./img/avatar-yeonuk.jpg" />
```
`alt`속성 좀 더 정확하게 적어주세요 :)

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+              <img alt="avatar" src="./img/avatar-yeonuk.jpg" />
+            </div>
+            <div class="friendsname">
+              <div class="names">
```
클래스 이름 좀 더 정확하게 적어주세요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+            </div>
+            <div class="friendsname">
+              <div class="names">
+                <span class="name">YeonukHwang</span>
```
클래스 이름 좀 더 정확하게 적어주세요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+                <span class="location">위워크 선릉 2호점</span>
+              </div>
+              <button>
+                <img alt="menubutton" src="./img/threedots.png" />
```
`alt` 속성에서는 이미지에 대한 설명을 더 잘 적어주세요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+            </div>
+          </header>
+          <div class="feedMain">
+            <img alt="feed image" src="./img/feedimg.jpeg" />
```
`alt`속성 자세하게 적어주세요! feed의 어떤 image인지 알 수가 없습니다!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+                  </div>
+                </li>
+                <li>
+                  <div class="commentListLeft">
```
class이름에 위치가 들어가는 건 좋지 않습니다!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+                    <a href="" class="name">YehuemChoi</a>
+                    <span>&nbsp코드를 한번 볼까요??</span>
+                  </div>
+                  <div class="commentListRight">
```
여기도 마찬가지! left, right 이런 위치는 클래스 이름에 적합하지 않습니다!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+                  </div>
+                  <div class="commentListRight">
+                    <img  class = "commentLikeBtn" alt="like-to-comment" src="./img/heart.png" />
+                    <button class = "commentDeleteBtn"><img alt="deletebutton" src="./img/del.png"></button>
```
`alt`속성에서 띄워쓰기 하셔도 됩니다!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+            </div>
+            <span class="timeToLastComment">42분 전</span>
+            <div class="writeComment">
+              <form onsubmit="return false;">
```
위에서와 마찬가지로 `preventDefault`를 검색해서 활용해보세요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+            <div class="avatar">
+              <img alt="avatar" src="./img/avatar-yeonuk.jpg" />
+            </div>
+            <div class="friendsname">
```
camelCase로 작성해주세요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+                <span class="location">서울시 어딘가</span>
+              </div>
+              <button>
+                <img src="./img/threedots.png" alt="menubutton" />
```
`alt`속성이 앞으로 오게 해주세요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+          </div>
+        </article>
+        <article class="feed">
+          <header class="feedHeader">
```
`header`태그는 주로 `h1`~`h6`태그를 사용할 때 사용합니다!
https://www.w3schools.com/tags/tag_header.asp

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+          </header>
+          <!-- right-storyList -->
+          <div class="storyContainer">
+            <div class="storyheader">
```
camelCase로 작성해 주세요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+                <div class="avatar">
+                  <img alt="avatar" src="./img/storypp1.jpg" />
+                </div>
+                <div class="friendsname">
```
camelCase로 작성해 주세요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+                <div class="avatar">
+                  <img alt="avatar" src="./img/storypp2.jpg" />
+                </div>
+                <div class="friendsname">
```
camelCase로 작성해 주세요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+                <div class="avatar">
+                  <img alt="avatar" src="./img/storypp3.jpg" />
+                </div>
+                <div class="friendsname">
```
camelCase로 작성해 주세요! 이제 안 적을게요!

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+                  <img alt="avatar" src="./img/storypp4.jpg" />
+                </div>
+                <div class="friendsname">
+                  <span class="name">Nick</span>
+                  <span class="description">2시간 전</span>
```
이 부분이 정중앙에 오지 않고 있는데 css 수정해주세요!

younuk23 replied:
description에 적용된 height속성, storycontainer 내부의 description에 대해서는 적용되지 않도록
height:inherit; 통해서 해결했습니다.

dipsiiiiiiiiii commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+          <!-- right-RecommendList -->
+          <div class="storyContainer">
+            <div class="storyheader">
+              <div class="storyname">회원님을 위한 추천</div>
+              <div class="seeAll">모두 보기</div>
```
`div`로 할 필요가 없을 것 같네요!

younuk23 replied:
span으로 변경했습니다.

younuk23 commented on students/11th/yeonukhwang/js/login.js:
```diff
@@ -0,0 +1,18 @@
+const button = document.getElementsByClassName("loginBtn")[0];
+const id = document.getElementById("id");
+const pw = document.getElementById("password");
+const form = document.getElementsByClassName("rightLogin")[0];
```
rightLogin class명 아이디 값을 받아오는 칸이라고 되어서 getId로 수정하겠습니다.

younuk23 commented on students/11th/yeonukhwang/js/login.js:
```diff
@@ -0,0 +1,18 @@
+const form = document.getElementsByClassName("rightLogin")[0];
+
+const changeBtnStatus = () => {
+  const idV = id.value;
+  const pwV = pw.value;
```
idValue, pwValue로 수정하겠습니다.

younuk23 commented on students/11th/yeonukhwang/js/login.js:
```diff
@@ -0,0 +1,18 @@
+const changeBtnStatus = () => {
+  const idV = id.value;
+  const pwV = pw.value;
+  if (idV.includes("@") && pwV.length >= 5) {
```
이메일 형식에 @로 나눠지는 부분 뒤쪽에 .com or .co .kr 등 '.'이 들어가는 부분이 있다면 email 형식인 것으로 check,
기능별로 validation 확인하는 함수 따로 선언하여서, changeBtnStatus안에서 호출하는 식으로 나눴습니다.

younuk23 commented on students/11th/yeonukhwang/js/login.js:
```diff
@@ -0,0 +1,18 @@
+const button = document.getElementsByClassName("loginBtn")[0];
```
loginBtn으로 수정하겠습니다.

younuk23 commented on students/11th/yeonukhwang/js/login.js:
```diff
@@ -0,0 +1,18 @@
+  const idV = id.value;
+  const pwV = pw.value;
+  if (idV.includes("@") && pwV.length >= 5) {
+    button.setAttribute("class", "loginBtn-active");
+    button.disabled = false;
+  } else {
+    button.setAttribute("class", "loginBtn");
+    button.disabled = true;
+  }
```
감사합니다 :)

younuk23 commented on students/11th/yeonukhwang/js/main.js:
```diff
@@ -0,0 +1,113 @@
+const article = Array.from(articlelist);
+
+// feed별 기능 구현
+article.forEach((ele) => {
```
수정했습니다 :)

younuk23 commented on students/11th/yeonukhwang/js/main.js:
```diff
@@ -0,0 +1,113 @@
+article.forEach((ele) => {
+  // 댓글기능
+  const ul = ele.getElementsByTagName("ul"); // get ul
+  const commentBtn = ele.getElementsByClassName("commentBtn")[0]; // get buttton
```
commentWriteBtn으로 수정하겠습니다.

younuk23 commented on students/11th/yeonukhwang/js/main.js:
```diff
@@ -0,0 +1,113 @@
+  // 댓글기능
+  const ul = ele.getElementsByTagName("ul"); // get ul
+  const commentBtn = ele.getElementsByClassName("commentBtn")[0]; // get buttton
+  const comment = ele.getElementsByClassName("comment")[0]; // get textarea
```
commentWritingArea로 변수명 및 class 명 변경하겠습니다.

younuk23 commented on students/11th/yeonukhwang/js/main.js:
```diff
@@ -0,0 +1,113 @@
+// feed별 기능 구현
+article.forEach((ele) => {
+  // 댓글기능
+  const ul = ele.getElementsByTagName("ul"); // get ul
```
넵 :)

younuk23 commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+    <script src="./js/main.js" defer></script>
+  </head>
+  <body>
+    <!-- header -->
```
tag명과 겹치는 부분의 주석은 삭제하겠습니다.

younuk23 commented on students/11th/yeonukhwang/js/main.js:
```diff
@@ -0,0 +1,113 @@
+// get feedlist to array
+const articlelist = document.getElementsByClassName("feed");
+const article = Array.from(articlelist);
```
dom api를 통해서 가져왔을때 prototype이 HTMLCollection으로 되어있어서 array메소드를 사용할 수 없어, Array.from을 통해 array형태로 만든 후에 forEach메소드를 사용했습니다.

younuk23 commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+            <img alt="feed image" src="./img/feedimg.jpeg" />
+          </div>
+          <div class="feedBottom">
+            <div class="bottomIconWrapper">
```
bottomIconWrapper -> feedBottomIconList 로 수정했습니다.

younuk23 commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+              </div>
+              <ul class="commentList">
+                <li>
+                  <div class="commentListLeft">
+                    <a href="" class="name">Yeonuk</a>
+                    <span>&nbsp이게 왜 안되지?</span>
+                  </div>
+                  <div class="commentListRight">
+                    <img  class = "commentLikeBtn" alt="like-to-comment" src="./img/heart.png" />
+                    <button class = "commentDeleteBtn">
+                      <img alt="deletebutton" src="./img/del.png">
+                    </button>
+                  </div>
```
comment파트(author name, comment)와 buttonpart(좋아요버튼, 삭제버튼)으로 두개 블락으로 나눠서 
dispaly:flex, justify-content:space-between 이용해서 양쪽 끝으로 위치시키려고 2개 div로 나눴습니다.
더 바람직한 방법 있으면 적용해서 수정하겠습니다.

younuk23 commented on students/11th/yeonukhwang/js/main.js:
```diff
@@ -0,0 +1,113 @@
+    commentListRight.appendChild(deleteBtn); //delbtn > rightdiv
+    commentbox.appendChild(commentListLeft); // leftdiv > li
+    commentbox.appendChild(commentListRight); // rightdiv > li
+    ul[0].appendChild(commentbox); // li > ul
```
적용해서 수정하겠습니다.

younuk23 commented on students/11th/yeonukhwang/js/main.js:
```diff
@@ -0,0 +1,113 @@
+  // - 댓글 추가 -
+  commentBtn.addEventListener("click", appendComment);
+  comment.addEventListener("keydown", (e) => {
+    let enter = e.keyCode ? e.keyCode : e.which;
```
초기 keycode로만 구현했을때, 페이지 로드 후 실행했을 때 가끔식 keycode인식 안되는 경우를 발견해서, keycode와 which 둘 다 사용해서, keycode값이 있을 경우 keycode값으로 적용하고, keycode값을 못 받았을 경우 which값 이용해서 값 설정하는 식으로 수정했습니다.

younuk23 commented on students/11th/yeonukhwang/js/main.js:
```diff
@@ -0,0 +1,113 @@
+    }); //
+
+    // - 추가된 댓글 좋아요 -
+    let commentClickCount = 0;
+    heart.addEventListener("click", () => {
+      commentClickCount++;
+      if (commentClickCount % 2 === 0) {
+        heart.setAttribute("src", "./img/heart.png");
+      }
+      if (commentClickCount % 2 === 1) {
+        heart.setAttribute("src", "./img/redheart.png");
+      }
+    });
```
```suggestion
    let isItLiked = false;
    heart.addEventListener("click", () => {
      if (!isItLiked) {
        heart.setAttribute("src", "./img/redheart.png");
        isItLiked = true;
      } else {
        heart.setAttribute("src", "./img/heart.png");
        isItLiked = false;
      }
    });
```
카운트횟수를 세는 방식이 아닌, 직관적인 기준인 기존에 좋아요가 눌렸었는지, 아닌지를 기준으로  수정하였습니다.

younuk23 commented on students/11th/yeonukhwang/login.html:
```diff
@@ -0,0 +1,52 @@
+        <div class="rightLogo">
+          <img alt="Instagram logo" src="./img/logo_text.png" />
+        </div>
+        <form class="rightLogin" onsubmit="return false;">
```
```suggestion
form.addEventListener("submit", (e) => {
  e.preventDefault();
});
```
위와 같이 수정했습니다.

younuk23 commented on students/11th/yeonukhwang/login.html:
```diff
@@ -0,0 +1,52 @@
+          <img alt="Instagram logo" src="./img/logo_text.png" />
+        </div>
+        <form class="rightLogin" onsubmit="return false;">
+          <div class="rightLoginItem">
```
리뷰해주신 코멘트 보고, div필요없다고 판단 삭제했습니다.

younuk23 commented on students/11th/yeonukhwang/main.html:
```diff
@@ -0,0 +1,446 @@
+            <img src="./img/logo_text.png" alt="Instagram-text-logo" />
+          </div>
+        </div>
+        <div class="search">
```
div 삭제했습니다.

saengmotmi commented:
리액트 공부 시작해주세요! 고생하셨습니다~

Commit:
Add; login 페이지 기능구현완료 및 main 레이아웃 작성중
- 로그인 레이아웃 완료
- 로그인 유효성검사 로직완료
- 메인 페이지 레이아웃 작성중

Commit:
Add : login 완료, main 레이아웃 진행중
- login 레이아웃 완료
- login inout tag에서 한 글자 이상 입력해야 버튼태그 작성가능
- 버튼 활성화되면 파란색으로 색 바뀜
- main 레이아웃 우측 사이드 진행중

Commit:
Add : main 우측 레이아웃 완료
- main 페이지 우측 레이아웃 완료

Commit:
Add : main layout 변경사항 추가
- 전체적으로 수정해서 커밋

Commit:
Add : 메인페이지 댓글기능 일부 추가
- 유저이름과 텍스트 댓글로 추가완료

Commit:
Add : 메인페이지 댓글기능 구현완료
- input tag에 글자 입력하면 닉네임(chunSig)이랑 해당 텍스트 댓글로 올라감
- 입력한 시간도 함수로 구현완

Commit:
Modify : 기존 코드로 백업
- img 태그로 사이즈 조절하려다가 오류나서 다시 수정

Commit:
Refactor : main css 스타일 속성 수정
- height 100%로 되어있던 부분 삭제

eunjeong-97 committed:
Update main.css
